ui/src/id.ts

```diff
@@ -0,0 +1,10 @@
+import * as org from "./org";
+import * as user from "./user";
+
+// Check if the given id is already taken
+export const isTaken = (id: string): Promise<boolean> =>
+  org.getOrg(id).then(org => !!org || user.get(id).then(user => !!user));
+
+// Check if the given id is available
+export const isAvailable = (id: string): Promise<boolean> =>
```

Comment:
We should consolidate these two and use `isAvailable` everywhere.

Reply:
If we can use `isAvailable` in a readable way in both site calls, sure! Thing is that at the moment, in one site call we need to check for `isAvailable` and in the other for `isTaken`. Having both is, in my view, nice given their current usage, it offers a more readable API for both scenarios, with no logic duplication. Happy to discuss it further!

Reply:
@xla in https://github.com/radicle-dev/radicle-upstream/pull/523/commits/b9d89bae19ec76676f1d8ed5688e7c0135a9c4f2 I have dry and clean this up by unifying the validation stores that were inconsistent across `OrgRegistration.svelte` and `PickHandleStep.svelte`. The caveat is that it also unifies terminology, now always mentioning "id". We can either make that depend on some input or we can simplify the language to be agnostic. For instance, we could:

- `Sorry, this id is already taken` -> `Sorry, this one is already taken` or `Sorry, it is already taken`
 - `Id is required` -> `This field is required`
- `Id should match (...)` -> `It should match (...)`

Reply:
Done in https://github.com/radicle-dev/radicle-upstream/pull/523/commits/b4f7bd6ea5f86d56963211605f6420d5c7818932